tests/persistAsync.test.tsx

```diff
@@ -44,7 +45,11 @@ describe('persist middleware with async configuration', () => {
     const storage = {
       getItem: async (name: string) =>
         JSON.stringify({
-          state: { count: 42, name },
+          state: {
+            count: 42,
+            name,
+            map: { type: 'Map', value: [['foo', 'bar']] },
+          },
```

Comment:
Could you create a new test case for Map and keep existing tests unmodified?

Reply:
Done, i pretty much copied the old persist- and rehyadrate tests and changed it to persist and rehydrate a map, I hope thats fine